pkg/lib/registry/registry_test.go

```diff
@@ -402,49 +400,42 @@ func TestCheckForBundles(t *testing.T) {
 						},
 					},
 					action: actionAdd,
-					expected: map[string]*registry.Package{
-						"testpkg": {
-							Name: "testpkg",
-							Channels: map[string]registry.Channel{
-								"alpha": {
-									Nodes: map[registry.BundleKey]map[registry.BundleKey]struct{}{
-										registry.BundleKey{
-											BundlePath: "unorderedReplaces-1.0.0",
-											Version:    "1.0.0",
-											CsvName:    "unorderedReplaces-1.0.0",
-										}: nil,
-										registry.BundleKey{
-											BundlePath: "unorderedReplaces-1.1.0",
-											Version:    "1.1.0",
-											CsvName:    "unorderedReplaces-1.1.0",
-										}: nil,
-										registry.BundleKey{
-											BundlePath: "unorderedReplaces-1.2.0",
-											Version:    "1.2.0",
-											CsvName:    "unorderedReplaces-1.2.0",
-										}: nil,
-									},
-								},
-								"stable": {
-									Nodes: map[registry.BundleKey]map[registry.BundleKey]struct{}{
-										registry.BundleKey{
-											BundlePath: "unorderedReplaces-1.0.0",
-											Version:    "1.0.0",
-											CsvName:    "unorderedReplaces-1.0.0",
-										}: nil,
-										registry.BundleKey{
-											BundlePath: "unorderedReplaces-1.1.0",
-											Version:    "1.1.0",
-											CsvName:    "unorderedReplaces-1.1.0",
-										}: nil,
-									},
-								},
+				},
+			},
+			wantErr: fmt.Errorf("add prunes bundle unorderedReplaces-1.0.0 (unorderedReplaces-1.0.0) from package testpkg, channel stable: this may be due to incorrect channel head (unorderedReplaces-1.1.0, skips/replaces [])"),
+		},
+		{
+			description: "silentPruneForExistingBundle",
+			steps: []step{
+				{
+					bundles: map[string]bundleDir{
+						"unorderedReplaces-1.0.0": {
+							csvSpec: json.RawMessage(`{"version":"1.0.0"}`),
+							annotations: registry.Annotations{
+								PackageName:        "testpkg",
+								Channels:           "stable,alpha",
+								DefaultChannelName: "stable",
+							},
+							version: "1.0.0",
+						},
+					},
+					action: actionAdd,
+				},
+				{
+					bundles: map[string]bundleDir{
+						"unorderedReplaces-1.2.0": {
+							csvSpec: json.RawMessage(`{"version":"1.2.0"}`),
+							annotations: registry.Annotations{
+								PackageName:        "testpkg",
+								Channels:           "alpha",
+								DefaultChannelName: "stable",
 							},
+							version: "1.2.0",
 						},
 					},
+					action: actionAdd,
```

Comment:
Probably outside the scope of this pull request, but it would be nice if we had a way of comparing the contents of the database based on these actions rather than just checking for error conditions.